src/Features/CSharp/Portable/SignatureHelp/PrimaryConstructorBaseTypeSignatureHelpProvider.cs

```diff
@@ -0,0 +1,153 @@
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            if (!TryGetBaseTypeSyntax(root, position, document.GetLanguageService<ISyntaxFactsService>(), triggerInfo.TriggerReason, cancellationToken, out var baseTypeSyntax))
+            {
+                return null;
+            }
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var within = (INamedTypeSymbol)semanticModel.GetDeclaredSymbol(baseTypeSyntax.Parent.Parent, cancellationToken);
+            var baseType = (INamedTypeSymbol)semanticModel.GetTypeInfo(baseTypeSyntax.Type, cancellationToken).Type;
+            if (within is null || baseType is null)
+            {
+                return null;
+            }
+
+            var accessibleConstructors = baseType.InstanceConstructors
+                .WhereAsArray(c => c.IsAccessibleWithin(within))
+                .WhereAsArray(c => c.IsEditorBrowsable(document.ShouldHideAdvancedMembers(), semanticModel.Compilation))
+                .Sort(semanticModel, baseTypeSyntax.SpanStart);
+
+            if (!accessibleConstructors.Any())
+            {
+                return null;
+            }
+
+            var anonymousTypeDisplayService = document.GetLanguageService<IAnonymousTypeDisplayService>();
+            var documentationCommentFormattingService = document.GetLanguageService<IDocumentationCommentFormattingService>();
+            var textSpan = SignatureHelpUtilities.GetSignatureHelpSpan(baseTypeSyntax.ArgumentList);
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+
+            var currentConstructor = semanticModel.GetSymbolInfo(baseTypeSyntax, cancellationToken).Symbol;
+            var selectedItem = TryGetSelectedIndex(accessibleConstructors, currentConstructor);
+
+            return CreateSignatureHelpItems(accessibleConstructors.SelectAsArray(c =>
+                Convert(c, baseTypeSyntax.ArgumentList.OpenParenToken, semanticModel, anonymousTypeDisplayService, documentationCommentFormattingService)).ToList(),
+                textSpan, GetCurrentArgumentState(root, position, syntaxFacts, textSpan, cancellationToken), selectedItem);
+        }
+
+        public override SignatureHelpState GetCurrentArgumentState(SyntaxNode root, int position, ISyntaxFactsService syntaxFacts, TextSpan currentSpan, CancellationToken cancellationToken)
+        {
+            if (TryGetBaseTypeSyntax(root, position, syntaxFacts, SignatureHelpTriggerReason.InvokeSignatureHelpCommand, cancellationToken, out var expression) &&
+                currentSpan.Start == SignatureHelpUtilities.GetSignatureHelpSpan(expression.ArgumentList).Start)
+            {
+                return SignatureHelpUtilities.GetSignatureHelpState(expression.ArgumentList, position);
+            }
+
+            return null;
+        }
```

Comment:
does this effectively mirror other sig help impls?

Reply:
Yes, this method was unmodified (from `ConstructorInitializerSignatureHelpProvider`)